src/transport/SecureSessionMgr.cpp

```diff
@@ -92,7 +92,21 @@ CHIP_ERROR SecureSessionMgr::SendMessage(NodeId peerNodeId, System::PacketBuffer
     return SendMessage(payloadHeader, peerNodeId, std::move(msgBuf));
 }
 
-CHIP_ERROR SecureSessionMgr::SendMessage(PayloadHeader & payloadHeader, NodeId peerNodeId, System::PacketBufferHandle msgBuf)
+CHIP_ERROR SecureSessionMgr::SendMessage(PayloadHeader & payloadHeader, NodeId peerNodeId, System::PacketBufferHandle msgBuf,
+                                         EncryptedPacketBufferHandle * bufferRetainSlot)
+{
+    return SendMessage(payloadHeader, peerNodeId, std::move(msgBuf), bufferRetainSlot, false, 0, 0);
+}
+
+CHIP_ERROR SecureSessionMgr::SendMessage(PayloadHeader & payloadHeader, NodeId peerNodeId, EncryptedPacketBufferHandle msgBuf,
+                                         EncryptedPacketBufferHandle * bufferRetainSlot, uint32_t msgIdIn, uint32_t payloadLenIn)
+{
+    return SendMessage(payloadHeader, peerNodeId, std::move(msgBuf), bufferRetainSlot, true, msgIdIn, payloadLenIn);
```

Comment:
I think we talked past each other a bit here.  This SendMessage overload can keep taking just an `EncryptedPacketBufferHandle` and not separate `msgIdIn` and `payloadLenIn` arguments, as long as the move constructor for `EncryptedPacketBufferHandle` copies those correctly.  But it should pass them separately to the `SendMessage` overload that takes just a `PacketBufferHandle` for the third arg.

So in other words, I'd take out the last two args of this `SendMessage` overload, and make it call:
```
     return SendMessage(payloadHeader, peerNodeId, std::move(msgBuf), bufferRetainSlot, true, msgBuf.GetMsgId(), msgBuf.GetPayloadLen());
```
or so.

Reply:
"return SendMessage(payloadHeader, peerNodeId, std::move(msgBuf), bufferRetainSlot, true, msgBuf.GetMsgId(), msgBuf.GetPayloadLen());"

This format assume the msgBuf.GetMsgId() and msgBuf.GetPayloadLen() are processed before std::move(msgBuf), right?  

Currently implementation can guarantee our code is generated in the absence of restrictions on expression evaluation order.

Reply:
But I think we can take out the last two args of this SendMessage in another format

Reply:
Removed the last two args of SendMessage API with EncryptedPacketBufferHandle passed in

Reply:
> This format assume the msgBuf.GetMsgId() and msgBuf.GetPayloadLen() are processed before std::move(msgBuf), right?

`std::move` on its own does nothing; it's just a cast to produce an rvalue reference.  But good catch; looks like construction of the argument from the rvalue reference is also non-deterministically sequenced with these evaluations.  So we want on-stack temporaries before the function call.